### PR TITLE
Prevent the extension from crashing when MediaSession API is not supported

### LINKED ISSUE #58: No longer playing in Opera for me getting error.
I get this error when trying to play in Opera Version:66.0.3515.72

Info

Uncaught TypeError: Cannot read property 'setActionHandler' of undefined

Context  

_generated_background_page.html

// Make notifyListeners public to allow for easier notification sending.
	window.notify = notifyListeners;

	if (DEBUG_FLAG) {
		window.setTime = function (hour, playTownTune) {
			notifyListeners("hourMusic", [hour, options.weather, options.music, playTownTune]);
		};
	}

}

### PR DESCRIPTION
**Pull request by jordigarcl**
Would close #58 

**TL;DR:** We accidentally broke support for other browsers (like Opera) when we used the `navigator.mediasession` API. This PR introduces checks around those calls so as only make them when said API is supported, therefore preventing the extension from crashing.

Tested in (macOS):
- Chrome
- Opera GX

Edited – tested in some more browsers:
- Brave
- Opera (vanilla)
- Vivaldi

Also, this is just a hotfix and does not fix the little glitches appearing e.g. in the icon badge. We should probably evaluate the extend of browser support.

### CODE REVIEW

**Comment by jordigarcl** [comment]
Please test this on Windows browsers. I wonder if we should have a "official" list of supported browsers so we can know what to test against.

File: scripts/background/AudioManager.js [hunk 1]
```diff
@@ -44,7 +44,9 @@ function AudioManager(addEventListener, isTownTune) {
 			} else playHourSong(game, weather, hour, false);
 		});
 
-		navigator.mediaSession.setActionHandler('nexttrack', null);
+		checkMediaSessionSupport(() => {
+			navigator.mediaSession.setActionHandler('nexttrack', null);
```
Comment (jordigarcl) [inline scripts/background/AudioManager.js:48]: A little fancier than just `if (supportsMediaSession) ** something() ** ;`

**Comment by jordigarcl** [comment]
If we are to release this as a patch ASAP, we can already set it as `4.0.1`and merge this directly into master (as a hotfix branch), then merge it to develop aswell.

**Comment by PikaDude** [comment]
IMO we'll go through develop first because there's a few other small things that I want to tinker with before the new patch. Additionally, a set list of browsers that we should test on before release sounds like a good idea. I was personally thinking:
- Google Chrome
- Opera
- Vivaldi
- Chromium-based Edge
- Maybe Brave?

I'll test this PR in all of these browsers in a few hours time.

**Comment by jordigarcl** [comment]
Tested the basic functionality in some more browsers, see PR description.

**Review by PikaDude — APPROVED** [review]

**Pull request merged**